packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -273,12 +270,12 @@ class VisualStudio {
   /// Returns the details dictionary for the newest version of Visual Studio.
   /// If [validateRequirements] is set, the search will be limited to versions
   /// that have all of the required workloads and components.
-  Map<String, dynamic> _visualStudioDetails({
+  Map<String, dynamic>? _visualStudioDetails({
       bool validateRequirements = false,
-      List<String> additionalArguments,
-      String requiredWorkload
+      List<String>? additionalArguments,
+      String? requiredWorkload
     }) {
-    final List<String> requirementArguments = validateRequirements
+    final List<String> requirementArguments = validateRequirements && requiredWorkload != null
```

Comment:
This isn't right; if we had a case where we didn't have a required workload, then only the first two things in the list should be removed, not the whole list. We should either assert that `requiredWorkload` is non-null if `validateArguments` is set, or restructure slightly so that we make a `['-requires', requiredWorkload]` array first and then conditionally include that in the list below.

(Obviously this was already wrong if that case happened; yay NNBD for finding latent bugs!)

Reply:
At the moment there isn't a code path that passes in `validateRequirements=true` but doesn't pass in `requiredWorkload` (I learned trying to write a test for this).  Unless you disagree I'm going to allow that combination instead of asserting since it seems possible you could just want to validate minimum versions of the build tools, cmake, etc.
```dart
    final List<String> requirementArguments = validateRequirements
        ? <String>[
            if (requiredWorkload != null) ...<String>[
              '-requires',
              requiredWorkload,
            ],
            ..._requiredComponents(_minimumSupportedVersion).keys
          ]
        : <String>[];
```